ENH: Synchronize changes between module widget and home widget

mohammadrashid0917 opened the pull request:
These changes make it so that changes in the module widget (viewable in Slicer) are reflected in the home widget (viewable in VR) and vice versa

mohammadrashid0917 committed:
ENH: Add VR optimized Qt Style Sheet

This style sheet makes the text in the VR menu widget bigger, and gives the buttons and slider handles different colors for unpressed, hover, and pressed statuses.

mohammadrashid0917 commented:
I'm not sure why the Add VR optimized Qt Style Sheet commit is showing up here. It only shows one commit with that name in my git log, not two. That commit should already be merged into this branch

If it's a problem could you try squashing as you merge? I'll keep trying to figure it out from my end until then

cpinter commented:
Please change the commit's name. It should contain what the change is, and not the activity you did to reach it. "ENH: Make VR home widget controls functional"

I think it shows up because you sent the PR to my fork and not the original repo.

cpinter commented on VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx:
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
+  d->LockMagnificationCheckBox->setEnabled(vrViewNode != nullptr);
+  d->LockMagnificationCheckBox->blockSignals(wasBlocked);
+  */
+  //Magnification lock of view node not implemented yet 
```
This is a todo, please make it so: //TODO: Magnification...

cpinter commented on VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx:
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
+
+  if (!vrViewNode)
+  {
+    qCritical() << Q_FUNC_INFO << " Failed: view node is null";
```
You also need to return here obviously, otherwise the next line results in an immediate crash.

cpinter commented on VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx:
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
+
+  if (!vrViewNode)
+  {
+    qCritical() << Q_FUNC_INFO << " Failed: view node is null";
```
Same here, please fix everywhere.

cpinter commented on VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx:
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
+  vrView->updateViewFromReferenceViewCamera();
+}
+*/
+//This member function won't work unless qSlicerVirtualRealityModule and qMRMLVirtualRealityView are included
```
Make it TODO

cpinter commented:
@mohammadrashid0917 I added some comments to the commit. Also please close this PR and create another one for KitwareMedical/SlicerVirtualReality instead of cpinter/SlicerVirtualReality

mohammadrashid0917 committed:
ENH: Make VR home widget controls functional

These changes make it so that changes in the module widget (viewable in Slicer) are reflected in the home widget (viewable in VR) and vice versa
KitwareMedical#43

cpinter commented on VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.h:
```diff
@@ -39,7 +39,7 @@ class qMRMLVirtualRealityHomeWidgetPrivate;
 {
   Q_OBJECT
-  QVTK_OBJECT
+    QVTK_OBJECT
```
Please undo this change and make QVTK_OBJECT the same indentation as Q_OBJECT
